### PR TITLE
Post SecureSessionEstablished event when secure session is established

### PR DESCRIPTION
**Pull request by wqx6**
We need a signal to notify that a secure session is  established.

### CODE REVIEW

File: src/protocols/secure_channel/PairingSession.cpp [hunk 1]
```diff
@@ -78,6 +81,20 @@ void PairingSession::Finish()
...
+        DeviceLayer::ChipDeviceEvent event;
+        event.Type                                   = DeviceLayer::DeviceEventType::kSecureSessionEstablished;
+        event.SecureSessionEstablished.TransportType = to_underlying(address.GetTransportType());
+        event.SecureSessionEstablished.SecureSessionType =
+            to_underlying(mSecureSessionHolder->AsSecureSession()->GetSecureSessionType());
+        if (mSecureSessionHolder->AsSecureSession()->GetSecureSessionType() == Transport::SecureSession::Type::kCASE)
+        {
+            event.SecureSessionEstablished.PeerNodeId  = mSecureSessionHolder->GetPeer().GetNodeId();
+            event.SecureSessionEstablished.FabricIndex = mSecureSessionHolder->GetPeer().GetFabricIndex();
+        }
```
Comment (andy31415) [inline src/protocols/secure_channel/PairingSession.cpp:93]: please ensure all fields are initialized. Even if not CASE, we should probably have known values for nodeid/fabriocindex.

File: src/protocols/secure_channel/PairingSession.cpp [hunk 2]
```diff
@@ -78,6 +81,20 @@ void PairingSession::Finish()
...
         VerifyOrDie(mSecureSessionHolder);
+        DeviceLayer::ChipDeviceEvent event;
+        event.Type                                   = DeviceLayer::DeviceEventType::kSecureSessionEstablished;
+        event.SecureSessionEstablished.TransportType = to_underlying(address.GetTransportType());
+        event.SecureSessionEstablished.SecureSessionType =
+            to_underlying(mSecureSessionHolder->AsSecureSession()->GetSecureSessionType());
+        if (mSecureSessionHolder->AsSecureSession()->GetSecureSessionType() == Transport::SecureSession::Type::kCASE)
+        {
+            event.SecureSessionEstablished.PeerNodeId  = mSecureSessionHolder->GetPeer().GetNodeId();
+            event.SecureSessionEstablished.FabricIndex = mSecureSessionHolder->GetPeer().GetFabricIndex();
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/PairingSession.cpp:92]: Nothing in the event definition clearly documents that these fields are not-to-be-looked-at for PASE sessions.

File: src/protocols/secure_channel/PairingSession.cpp (hunk 1 above)
Comment (wqx6) [reply src/protocols/secure_channel/PairingSession.cpp]: Done

File: src/protocols/secure_channel/PairingSession.cpp (hunk 2 above)
Comment (wqx6) [reply src/protocols/secure_channel/PairingSession.cpp]: Done

File: src/protocols/secure_channel/PairingSession.cpp [hunk 3]
```diff
@@ -78,6 +81,17 @@ void PairingSession::Finish()
     if (err == CHIP_NO_ERROR)
     {
         VerifyOrDie(mSecureSessionHolder);
+        DeviceLayer::ChipDeviceEvent event;
+        event.Type                                   = DeviceLayer::DeviceEventType::kSecureSessionEstablished;
+        event.SecureSessionEstablished.TransportType = to_underlying(address.GetTransportType());
+        event.SecureSessionEstablished.SecureSessionType =
+            to_underlying(mSecureSessionHolder->AsSecureSession()->GetSecureSessionType());
+        event.SecureSessionEstablished.PeerNodeId  = mSecureSessionHolder->GetPeer().GetNodeId();
+        event.SecureSessionEstablished.FabricIndex = mSecureSessionHolder->GetPeer().GetFabricIndex();
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/PairingSession.cpp:90]: Might be good to include the local session id here too?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/protocols/secure_channel/PairingSession.cpp (hunk 3 above)
Comment (wqx6) [reply src/protocols/secure_channel/PairingSession.cpp]: Done.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**